miden-tx/src/compiler/tests.rs

```diff
@@ -248,7 +245,7 @@ fn create_note_leafs(tx_compiler: &mut TransactionComplier, note: &Note) -> Code
         .assembler
         .compile_in_context(
             note.script().code(),
-            &mut AssemblyContext::new(assembly::AssemblyContextType::Program),
+            &mut AssemblyContext::for_program(&note.script().code),
```

Comment:
nit: on line 248 we should probably use the same `code()` function as on line 247 - i.e., `&note.script().code()`.